.github/workflows/integration-testing.yml

```diff
@@ -2,6 +2,14 @@ name: Integration Testing
 on:
   push:
     branches: [ main ]
+  workflow_dispatch:
+    inputs:
+      push: 
+        description: 'push docker image'
+        type: boolean
+        required: true
+  schedule:
+    - cron: '0 11 * * Mon'
```

Comment:
One complication here is that only the `push to main` job pushes a new image. So if the `push to main` job fails, then the schedule will keep trying to pull an image that never got published at the validation step.

Same problem with the `workflow_dispatch`.

Do you think we should have an extra step to validate whether the docker Image we want to use in the Integration Tests actually exists before we run the Validation Steps? I'm okay with not doing it either, to simplify the workflow here.

Reply:
I expect the schedule job never push new image, it is just verify the latest dependencies.

Reply:
I see. We can push new image for schedule job weekly.